docs/edgeql/overview.rst

```diff
@@ -229,3 +229,74 @@ A query is evaluated recursively using the following procedure:
        the results of the invocations are collected to form a single set.
 
 6. Collect the results of all iterations to obtain the final result set.
+
+
+.. _ref_eql_polymorphic_queries:
+
+Polymorphic Queries
+===================
+
+:index: nested shapes
+
+Sometimes it is useful to have an abstract type as a link target, thus
+allowing objects of different extending types to be referenced.  This
+necessitates writing *polymorphic queries* that could fetch different
+data depending on the type of the actual objects.  Consider the
+following schema:
+
+.. code-block:: sdl
+
+    abstract type Named {
+        required property name -> str {
+            delegated constraint exclusive;
+        }
+    }
+
+    type User extending Named {
+        property avatar -> str;
+        multi link favorites -> Named;
+    }
+
+    type Game extending Named {
+        property price -> int64;
+    }
+
+    type Article extending Named {
+        property url -> str;
+    }
+
+The above schema describes some platform where there are users, games
+and articles.  Each ``User`` can add any of these entities to
+``favorites``.  A natural question to ask then is how to fetch all this
+diverse data in a single query:
+
+.. code-block:: edgeql
+
+    SELECT User {
+        name,
+        avatar,
+        favorites: {
+            # common to all Named
+            name,
+
+            # specific to Games
+            [IS Game].price,
+
+            # specific to Article
+            [IS Article].url,
+
+            # specific to User
+            [IS Game].avatar,
+
+            # a computable value tracking how many favorites
+            # does my favorite User have?
+            favorites_count := count(
+                # start the path at the root of the shape
+                User.favorites[IS User].favorites)
+        }
+    }
+
+The :eql:op:`[IS TypeName] <ISINTERSECT>` construct can be used in
+:ref:`paths <ref_eql_expr_paths>` to restrict the target to a specific
+type.  When it is used in :ref:`shapes <ref_eql_expr_shapes>` it
```

Comment:
"When it is used" sentence is redundant here. You literally just showed that 20 words earlier.

Reply:
This is deliberate. It is the first time the `[IS type]` is used or mentioned. Given that this is the "overview" section there's no reason to believe that the reader is already familiar with it. So the paragraph is meant to reinforce the idea that there are, in fact, two different usages of this syntax (both of which appear in the example).